x-pack/platform/plugins/shared/streams/server/routes/internal/streams/processing/simulation_handler.ts

```diff
@@ -234,40 +272,35 @@ const preparePipelineSimulationBody = (
 
 const prepareIngestSimulationBody = (
   simulationData: ReturnType<typeof prepareSimulationData>,
+  streamIndex: IndicesIndexState,
   params: ProcessingSimulationParams
-) => {
-  const { path, body } = params;
+): SimulateIngestRequest => {
+  const { body } = params;
   const { detected_fields } = body;
 
   const { docs, processors } = simulationData;
 
-  // TODO: update type once Kibana updates to elasticsearch-js 8.17
-  const simulationBody: {
-    docs: IngestDocument[];
-    pipeline_substitutions: Record<string, IngestPipelineConfig>;
-    component_template_substitutions?: Record<string, ClusterComponentTemplateNode>;
-  } = {
+  const defaultPipelineName = streamIndex.settings?.index?.default_pipeline;
+  const mappings = streamIndex.mappings;
+
+  const simulationBody: SimulateIngestRequest = {
     docs,
-    pipeline_substitutions: {
-      [`${path.name}@stream.processing`]: {
-        processors,
+    ...(defaultPipelineName && {
+      pipeline_substitutions: {
+        [defaultPipelineName]: {
+          processors,
+        },
+      },
+    }),
+    mapping_addition: {
+      ...mappings,
+      properties: {
+        ...(mappings && mappings.properties),
```

Comment:
Could we add a comment here explaining that there is some weirdness around this behavior and that it might change? Also, we should add an api integration test for it so we don't have Elasticsearch "fix" this in a year or so and break this.

Reply:
Addressed with [test(streams): add test for field mapping failures](https://github.com/elastic/kibana/pull/216877/commits/58eb01c8bd03f23097539541abcebb692985fde3)